packages/trie/src/db/checkpoint.ts

```diff
@@ -177,12 +178,12 @@ export class CheckpointDB implements DB {
   async del(key: Uint8Array): Promise<void> {
     const keyHex = bytesToHex(key)
     if (this._cache !== undefined) {
-      this._cache.set(keyHex, null)
```

Comment:
This will likely not fully work. There are different semantics in the cache for „known to not exist“ (as meant here) which was null before and „not in the cache“ which was already undefined before, I battled with this a long time.

This comes into play when the cache get method is called, because this gives undefined when the key is not in the cache, now it also gives undefined for the above case (so there is this together-throwing).

This is why I added a somewhat more complex dict structure in the account cache for the account value or undefined, so there is an „outer“ and an „inner“ undefined, which makes up for the differentiation there.

Maybe it is easier to keep null here, I was actually never that happy with the account cache structure regarding this, since it creates some overhead.

Reply:
Understood, I haven't really dug into the `trie` piece of it yet.  I've been focused on trying to get `blockchain` passing all the tests while removing as much of the dependence on `try/catch` logic as possible.